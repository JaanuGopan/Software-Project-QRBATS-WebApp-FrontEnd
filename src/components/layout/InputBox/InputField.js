import React from "react";
import "./inputfield.css";

const InputField = ({ placeholder, value, onChange, inputType }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className="input-field"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default InputField;
