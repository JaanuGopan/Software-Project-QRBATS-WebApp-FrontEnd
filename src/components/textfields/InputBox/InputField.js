import React from "react";
import "../../../pages/Event/EventCreation/EventCreation.css";

const InputField = ({ placeholder, value, onChange, inputType }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className="form-control mb-2"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default InputField;
