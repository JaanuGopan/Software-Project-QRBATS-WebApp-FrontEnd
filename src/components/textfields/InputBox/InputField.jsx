import React from 'react';
import '../../../pages/Event/EventCreation/EventCreation.css';

const InputField = ({ placeholder, value, onChange, inputType, error, helperText }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className={`form-control mb-2 ${error ? 'is-invalid' : ''}`}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
      {error && <div className="invalid-feedback">{helperText}</div>}
    </div>
  );
};

export default InputField;
