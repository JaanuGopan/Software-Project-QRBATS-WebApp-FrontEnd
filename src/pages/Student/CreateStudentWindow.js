import React, { useState } from "react";
import "./Student.css";
import axios from "axios";
import Designer from "../../assets/Images/Designer.jpeg";
import { useNavigate } from "react-router-dom";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";

const CreateStudentWindow = ({ handlecloseCreateStudentWindow }) => {
  const [firstname, setFirstName] = useState("");
  const [lastname, setLastName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [userName, setUserName] = useState("");
  const [userRole, setUserRole] = useState("");
  const [departmentId, setDepartmentId] = useState("");
  const navigate = useNavigate();

  const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];
  const userRoleList = ["ADMIN", "LECTURER", "STAFF"];

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await axios.post(
        "http://localhost:8080/api/v1/auth/signin",
        {
          firstName: firstname,
          lastName: lastname,
          email: email,
          password: password,
          userName: userName,
          departmentId: deparmentList.indexOf(departmentId) + 1,
        }
      );
      const token = response.data.token;
      const decodedToken = parseJwt(token);
      const userName = decodedToken.sub;
      localStorage.setItem("token", token);
      navigate("/mainNavigation", { state: { userName } });
    } catch (error) {
      console.error("Login failed", error);
    }
  };

  const parseJwt = (token) => {
    const base64Url = token.split(".")[1];
    const base64 = base64Url.replace(/-/g, "+").replace(/_/g, "/");
    const jsonPayload = decodeURIComponent(
      atob(base64)
        .split("")
        .map((c) => "%" + ("00" + c.charCodeAt(0).toString(16)).slice(-2))
        .join("")
    );
    return JSON.parse(jsonPayload);
  };

  return (
    <div className="student-signup-main-container">
      <div
        className="closeCreateEventWindow"
        onClick={handlecloseCreateStudentWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <p className="student-head1">Create Student</p>
      <div className="student-login-container">
        <div className="student-image-container">
          <img src={Designer} className="student-logo" alt="Logo" />
        </div>
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            {/* First Name Input */}
            <InputField
              placeholder="Enter your first name"
              value={firstname}
              onChange={(e) => setFirstName(e.target.value)}
              inputType="text"
            />

            {/* Last Name Input */}
            <InputField
              placeholder="Enter your last name"
              value={lastname}
              onChange={(e) => setLastName(e.target.value)}
              inputType="text"
            />

            {/* Email Input */}
            <InputField
              placeholder="Enter your email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              inputType="text"
            />

            <InputField
              placeholder="Enter your user name"
              value={userName}
              onChange={(e) => setUserName(e.target.value)}
              inputType="text"
            />

            {/* Password Input */}
            <InputField
              placeholder="Enter your password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              inputType="password"
            />

            <div className="choice-input mb-3">
              <select
                value={departmentId}
                onChange={(e) => setDepartmentId(e.target.value)}
                className="student-select-input"
              >
                <option value="">Select the department</option>
                {deparmentList.map((option, index) => (
                  <option key={index} value={option}>
                    {option}
                  </option>
                ))}
              </select>
            </div>
            <button type="submit" className="btn btn-primary w-100">
              Add Student
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default CreateStudentWindow;
