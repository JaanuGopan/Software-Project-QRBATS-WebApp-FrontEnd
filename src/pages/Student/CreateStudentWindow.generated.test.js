import renderer from 'react-test-renderer';
import React, { useState } from "react";

import axios from "axios";
import Designer from "../../assets/Images/Designer.jpeg";
import { useNavigate } from "react-router-dom";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";
import CreateStudentWindow from "./CreateStudentWindow";

jest.mock("./Student.css");
jest.mock("axios");
jest.mock("../../assets/Images/Designer.jpeg");
jest.mock("react-router-dom");
jest.mock("../../components/textfields/InputBox/InputField");
jest.mock("react-icons/io");

const renderTree = tree => renderer.create(tree);
describe('<CreateStudentWindow>', () => {
  it('should render component', () => {
    expect(renderTree(<CreateStudentWindow 
    />).toJSON()).toMatchSnapshot();
  });
  
});