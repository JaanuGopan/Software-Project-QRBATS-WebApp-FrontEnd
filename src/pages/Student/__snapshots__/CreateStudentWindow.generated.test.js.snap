// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateStudentWindow> should render component 1`] = `
<div
  className="student-signup-main-container"
>
  <div
    className="closeCreateEventWindow"
  />
  <p
    className="student-head1"
  >
    Create Student
  </p>
  <div
    className="student-login-container"
  >
    <div
      className="student-image-container"
    >
      <img
        alt="Logo"
        className="student-logo"
        src="Designer.jpeg"
      />
    </div>
    <div
      className="form-container"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="eventCreation-form"
        >
          <select
            className="form-control mb-2"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Select the department
            </option>
            <option
              value="DEIE"
            >
              DEIE
            </option>
            <option
              value="DCOM"
            >
              DCOM
            </option>
            <option
              value="DMME"
            >
              DMME
            </option>
            <option
              value="DCEE"
            >
              DCEE
            </option>
            <option
              value="DMENA"
            >
              DMENA
            </option>
          </select>
        </div>
        <button
          className="btn btn-primary w-100"
          type="submit"
        >
          Add Student
        </button>
      </form>
    </div>
  </div>
</div>
`;
